internal/util/reftracker/radoswrapper/interface.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2022 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package radoswrapper
+
+import (
+	"github.com/ceph/go-ceph/rados"
+)
+
+// These interfaces are just wrappers around some of go-ceph's rados pkg
+// structures and functions. They have two implementations: the "real" one
+// (that simply uses go-ceph), and a fake one, used in unit tests.
+
+// IOContextW is a wrapper around rados.IOContext.
+type IOContextW interface {
+	// GetLastVersion will return the version number of the last object read or
+	// written to.
+	GetLastVersion() (uint64, error)
+
+	// GetXattr gets an xattr with key `name`, it returns the length of
+	// the key read or an error if not successful
+	GetXattr(oid string, key string, data []byte) (int, error)
+
+	// CreateWriteOp returns a newly constructed write operation.
+	CreateWriteOp() WriteOpW
+
+	// CreateReadOp returns a newly constructed read operation.
+	CreateReadOp() ReadOpW
+}
+
+// WriteOpW is a wrapper around rados.WriteOp interface.
+type WriteOpW interface {
+	// Create a rados object.
+	Create(exclusive rados.CreateOption)
+
+	// Remove object.
+	Remove()
+
+	// SetXattr sets an xattr.
+	SetXattr(name string, value []byte)
+
+	// WriteFull writes a given byte slice as the whole object,
+	// atomically replacing it.
+	WriteFull(b []byte)
+
+	// SetOmap appends the map `pairs` to the omap `oid`.
+	SetOmap(pairs map[string][]byte)
+
+	// RmOmapKeys removes the specified `keys` from the omap `oid`.
```

Comment:
should have been explicit with the name like `RemoveOmapKeys` , this (`RmOmap..`) is close to convey the meaning though.

Reply:
All these methods are actually just copy&paste of go-ceph's RADOS interface ([`IOContext.RmOmapKeys`](https://github.com/ceph/go-ceph/blob/6a4d7eb0bdf7872d0890bfaf2a3de70c05d892ed/rados/omap.go#L191-L197)) -- I wanted to keep them exactly as they are in the library since this is just a wrapper. Let me know if this makes sense and is ok to keep it, or you want me to change it.

Reply:
sure, lets keep with the same/existing name then :+1: